### PR TITLE
pai expansion cards

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
lets some circuit boards be installed into pais to let them do more things

currently its criminal records, which is read-only unless you emag it (new thief meta to lay low)

## Why / Balance
https://github.com/space-wizards/docs/pull/183

## Technical details
added new `PAIExpansionSlot` and `PAIExpansionCard` components, uses componentregistry gaming and requires engine pr
## Media
(the examine, no longer available for power monitor)

criminal records is real

radio shows pai name and unknown officer so its clear its emagged, you need to microwave it to not get outed (set to syndicate for testing just to see it)

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**
:cl:
- add: Certain computer boards can be inserted into pAIs to give them extra features, examine them to see if they are allowed.

### CODE REVIEW

File: Resources/Textures/Interface/Actions/pai-expansions.rsi/power_monitor.png
Comment (lzk228) [inline Resources/Textures/Interface/Actions/pai-expansions.rsi/power_monitor.png:1]: maybe we can combine that with aghost actions icons?

**Comment by superjj18** [comment]
I do very much love this change.  I had a similar idea, however the only difference was that instead of expansion slots, its connected to the unique command PDAs, and give access once the Pai is inserted.  This allows not only the balancing of only one pai on the station having access to their specific department utilities, and also allows for antag strats involving stealing a command pda for their syndie pai(they would likely keep their original pda for their uplink, which in turn can make them vulnerable to random sec searches)

File: Resources/Textures/Interface/Actions/pai-expansions.rsi/power_monitor.png
Comment (deltanedas) [reply Resources/Textures/Interface/Actions/pai-expansions.rsi/power_monitor.png:1]: pro idea give aghost expansion slots :trollface:

**Comment by UbaserB** [comment]
Is this going to be obtained through research?

**Comment by deltanedas** [comment]
no theyre just computer circuitboards you put in the pai

**Comment by UbaserB** [comment]
> no theyre just computer circuitboards you put in the pai

yea but like how do you get them

**Comment by Dutch-VanDerLinde** [comment]
> > no theyre just computer circuitboards you put in the pai
> 
> yea but like how do you get them

they're just normal circuit boards for any normal computer

**Comment by UbaserB** [comment]
> > > no theyre just computer circuitboards you put in the pai
> > 
> > 
> > yea but like how do you get them
> 
> they're just normal circuit boards for any normal computer

wait im dumb

**Comment by superjj18** [comment]
> > > no theyre just computer circuitboards you put in the pai
> > 
> > 
> > yea but like how do you get them
> 
> they're just normal circuit boards for any normal computer

big thing here would be balancing, as not every single person should be able to get a pai with access to say a crew monitor for instance.

**Comment by deltanedas** [comment]
they need a crew monitor to begin with, they can always just drag around a computer/ the parts for it wherever they go. the real power is having another player keep track of it and not yourself

**Comment by chromiumboy** [comment]
I think you'll have trouble getting this PR accepted, as your converting consoles into portable versions of themselves, and this has been vetoed in the past

**Comment by lzk228** [comment]
> I think you'll have trouble getting this PR accepted, as your converting consoles into portable versions of themselves, and this has been vetoed in the past

but you cant see it directly, pai will tell you about the maps and etc

File: Resources/Prototypes/Entities/Objects/Fun/pai.yml [hunk 1]
```diff
@@ -27,6 +27,21 @@
   - type: Input
     context: "human"
   - type: PAI
+  - type: PAIExpansionSlot
+  - type: WiresPanel
+  - type: ItemSlots
+    slots:
+      expansion:
+        name: Expansion Slot
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Objects/Fun/pai.yml:35]: loc string

**Review by lzk228 — CHANGES_REQUESTED** [review]

**Comment by chromiumboy** [comment]
> > I think you'll have trouble getting this PR accepted, as your converting consoles into portable versions of themselves, and this has been vetoed in the past
> 
> but you cant see it directly, pai will tell you about the maps and etc

Ah I see. Hmm, OK, that seems better then

**Comment by mbreton101** [comment]
This has a lot of potential with helping all departments making their job a little easier, it would also allow for Pai's to get less bord. here's some examples why this could be cool:
Allows Paramedics to get notifications of suit sensors on the move, without needing the handheld
Allows Security officer to set people to arrest easier
Allows Cargo to order delivery's anywhere 
Allows Science to research artifacts using a Pai potentially allowing more roleplay, or monitor/adjusting science's technology's

This could also potently give syndicate Pai a unique benefits over normal Pai's. Either having multiple expansion slots allowing syndicates to monitor multiple departments or potentially allowing them to access security records or approving orders without the proper access.

I think in order to allow the change of Cargo/security computer's, their should be an ID slot that you can put in the Pai, to give them access to change restricted terminals. I don't think they ID needs to be stuck in the Pai, just in an out.

Lastly my experimental idea would allow a Pai to change access of crew members, granted they get an ID computer installed, and have given the Pai proper HoP/Captain access. This would allow the HoP to not be completely bounded to their office, allowing for more unique playstyles. I personally believe this will not majorly benefits an-tag's, because an an-tag would both need to get an ID computer circuit and HoP/Captain access; and if an an-tag already has those items they practically they already have all access

**Comment by SoulFN** [comment]
What about inserting encryption keys into PAI?

**Comment by Golinth** [comment]
> I think in order to allow the change of Cargo/security computer's, their should be an ID slot that you can put in the Pai, to give them access to change restricted terminals. I don't think they ID needs to be stuck in the Pai, just in an out.

instead, why not have pAIs inherit the access of the PDA they’re inserted in?

**Comment by deltanedas** [comment]
pai yelling on common that owner is being killed by john smith the tator

**Comment by HoidC** [comment]
> pai yelling on common that owner is being killed by john smith the tator

let them read but not send on radio

**Comment by SoulFN** [comment]
Can you also add PAI anomaly scanner and gas analyser functionality, but only PAI can see information?

**Comment by deltanedas** [comment]
> Can you also add PAI anomaly scanner and gas analyser functionality, but only PAI can see information?

they arent boards so wouldnt make sense to insert them, and theres no cognitive load on having an anomaly scanner open 24/7 so it doesn't serve much purpose to have a pai look at the funny numbers for the shift compared to something more interesting like a crew monitor

**Comment by Agustron24** [comment]
Syndicate pai should be able to crack doors, like them beging able to pulse cables of doors.

Also make pai have sake game on them

**Comment by Agustron24** [comment]
*snake

**Comment by Emisse** [comment]
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

**Comment by deltanedas** [comment]
open

**Comment by deltanedas** [comment]
how does it have different engine on actions than i have locally

**Comment by deltanedas** [comment]
@AJCM-git do you know why its using wrong engine code just for this pr

**Comment by AJCM-git** [comment]
> @AJCM-git do you know why its using wrong engine code just for this pr

How do you even run into things like this (pretty sure when it merges master with your required pr, it autoresolves the conflicts by adding another `SetUi` with a repeated signature in the runner version of the engine)

image

**Comment by IProduceWidgets** [comment]
> criminal records, which is read-only unless you emag it (new thief meta to lay low)
By chance did you check if the ninja obj is impacted by this?

**Comment by deltanedas** [comment]
> By chance did you check if the ninja obj is impacted by this?

you need the console board to make a pai a criminal records console so no real reason to not allow hacking it

**Comment by deltanedas** [comment]
OH pr body was trolling it real

**Comment by metalgearsloth** [comment]
This seems like significant powercreep for PAIs by being able to see everyone with a crew monitor, and also power monitoring is incredibly expensive so in no universe should they have this.

**Comment by deltanedas** [comment]
do you have any suggestions besides criminal records then

**Comment by keronshb** [comment]
> do you have any suggestions besides criminal records then

PAIs should really be nothing more than a glorified talking radio that can change their holographic shape, and even then I have doubts for that because they've ruined intrigue rounds in SS13 because of the fact they're hidden sentient pocket radios.

**Comment by deltanedas** [comment]
well the design doc was merged :trollface:

**Comment by deltanedas** [comment]
@metalgearsloth no funny ones now

**Comment by Djungelskog2** [comment]
This partially feels like a more convoluted attempt to add more PDA cartridges to the game, also I take issue with the lathe and announcement ones because if in missing something they can be both easily obtained and abused, they can all be obtained from things like tech vaults or science and can be used to constantly recall the shuttle, spam announcements, store absurd amounts of mats or make anything anywhere you go

**Comment by Djungelskog2** [comment]
I would rather have PDA cartridges locked behind something like science rather than the exact same functionality with the liability of a player behind it. I like the idea of it being a personal assistant with an alternative functionality rather than an upgraded or new functionality (think hacking doors in 13, or adding holograms to make it more flavourful and interactive)

**Comment by Baptr0b0t** [comment]
> This partially feels like a more convoluted attempt to add more PDA cartridges to the game, also I take issue with the lathe and announcement ones because if in missing something they can be both easily obtained and abused, they can all be obtained from things like tech vaults or science and can be used to constantly recall the shuttle, spam announcements, store absurd amounts of mats or make anything anywhere you go

it's true, however it's not necessarily easy to obtain and it's not that different from creating a computer

**Comment by superjj18** [comment]
I still think these additional abilities should be connected to the Command PDAs for balance reasons.  Makes command members with pais slightly better and limits the amount of pais with abilities to characters that arguably should have them.

(Also can be stolen and used by syndie pais if the command member is killed)

**Comment by Emisse** [comment]
i think it might be better to allow pai to interact with anything installed in your pda

**Comment by Jezithyr** [comment]
> well the design doc was merged
The design proposal was reverted as its approval did not take into account the discussions or balance/powercreep issues.

> i think it might be better to allow pai to interact with anything installed in your pda
This is a far better solution and doesn't add additional maintenance of a separate upgrade system. Plus it's far more balanced since the PAI can only do what a player would otherwise do themselves on their PDA, instead of having additional functionality which would need to be balanced and maintained (which adds more risks of creating bugs/exploits).

**Comment by superjj18** [comment]
> i think it might be better to allow pai to interact with anything installed in your pda

If PDA messaging gets added this honestly would be pretty cool

**Pull request closed without merging**